daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Converter.scala

```diff
@@ -162,7 +162,7 @@ object Converter {
   }
 
   private def fromAnyChoice(
-      lookupChoice: (Identifier, Name) => Either[String, TemplateChoiceSignature],
+      lookupChoice: (Identifier, Name) => Either[String, GenTemplateChoice[_]],
```

Comment:
You actually cannot do more with `GenTemplateChoice[_]` that you can do with `TemplateChoiceSignature = TemplateChoice[Unit]`.

It is a bit annoying here, but the change is due to  `Interface` returning `GenXxxx[_]` and not `XxxSignature`.
There is basically two reason for that: 
- Interface does not have to work with Signature it is just a wrapper around `GenPackage[_]`, so we can use it with package where value are substituted with Unit or not. (It is more flexible like).
- The only place where PackageSignature is very usefull is inside CompiledPackageto be sure we do not store the values. 
- Even if we change the type of Interface to take PackageSignature instead of GenPackage[_] we still have some issue with pattern matching and  type erasure in to return GenXxxx[Unit] in some of the lookup (e.g.  `lookupValue`).

I am considering one of the two changes (for an upcomming PR)
1. remove all The `XxxSignature` type aliases (exception `PackageSignature`) which are now a pretty useless
2. redefined all the `XxxSignature` to be alias of `GenXxx[_]` and defined a use directly `GenPackage[Unit]` in `CompiledPackage`

Reply:
Just to be clear, I don’t think this is merge-blocking but at least something to think about and potentially address later.

Reply:
Making Interface generic is quite a pain. 
I try again to make it use `PackageSignature` and everything went smooth this time.
7b6818ea0a1ef5d866cc9ee4d2021c7eeb4a0fd4
This definitively prevent the usage of lookup in the `ScenarioLoader` as explained below but I think it not worthwhile to generalize `Interface` for only one use case.